### PR TITLE
Fix limit on the commit retriever

### PR DESCRIPTION
**Pull request by Lothiraldan**
It should works in the integration test case, adding a warning if the argument
is bypassed to detect the cases where it doesn't works.

### CODE REVIEW

**Comment by Lothiraldan** [comment]
I've forced the commit retriever to not download the existing DB but now I'm struggling to find which amount of recent commits to analyze for making all the scripts happy.

Here is the traceback with the last 2000 commits:

```
2019-09-27 19:58:08,661:INFO:bugbug.repository:Pulling and updating mozilla-central
2019-09-27 19:58:11,199:INFO:bugbug.repository:mozilla-central pulled and updated
Prev version 3
Current version 3
Getting revs from -2000 to tip...
Mining 1966 commits using 4 processes...
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5/5 [00:11<00:00,  2.31s/it]
Downloading file->component mapping...
Mining 1966 commits using 4 processes...
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1966/1966 [07:28<00:00,  4.38it/s]
Analyzing experiences from 1966 commits...
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1966/1966 [00:00<00:00, 872408.13it/s]
  0%|                                                                                                                                                                                       | 0/1966 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "/home/lothiraldan/.virtualenvs/mozilla-automl-3.7/bin/bugbug-data-commits", line 11, in <module>
    load_entry_point('bugbug', 'console_scripts', 'bugbug-data-commits')()
  File "/home/lothiraldan/project/mozilla-automl/bugbug/scripts/commit_retriever.py", line 65, in main
    retriever.retrieve_commits(args.limit)
  File "/home/lothiraldan/project/mozilla-automl/bugbug/scripts/commit_retriever.py", line 42, in retrieve_commits
    repository.download_commits(self.repo_dir, rev_start)
  File "/home/lothiraldan/project/mozilla-automl/bugbug/bugbug/repository.py", line 643, in download_commits
    calculate_experiences(commits, first_pushdate, save)
  File "/home/lothiraldan/project/mozilla-automl/bugbug/bugbug/repository.py", line 522, in calculate_experiences
    update_experiences("author", day, [commit.author])
  File "/home/lothiraldan/project/mozilla-automl/bugbug/bugbug/repository.py", line 395, in update_experiences
    for exp, item in zip(total_exps, items)
  File "/home/lothiraldan/project/mozilla-automl/bugbug/bugbug/repository.py", line 395, in <listcomp>
    for exp, item in zip(total_exps, items)
  File "/home/lothiraldan/project/mozilla-automl/bugbug/bugbug/repository.py", line 382, in get_experience
    return experiences[exp_type][commit_type][item][day]
  File "/home/lothiraldan/project/mozilla-automl/bugbug/bugbug/utils.py", line 221, in __getitem__
    ), f"Can't get a day ({day}) from earlier than start day ({self.start_day})"
AssertionError: Can't get a day (4106) from earlier than start day (4110)

```

**Comment by marco-c** [comment]
It looks like it's failing when calculating the experience for the first commit (`0/1966`), this made me think it could be caused by the loading of an earlier commit_experiences, but looking at your patch that doesn't seem to be possible...
Could you check just to make sure that we hit https://github.com/mozilla/bugbug/blob/cd6d8dc2c39355f46cb24b2f1e9386ce92ad47b0/bugbug/repository.py#L354 and not https://github.com/mozilla/bugbug/blob/cd6d8dc2c39355f46cb24b2f1e9386ce92ad47b0/bugbug/repository.py#L352?

**Comment by Lothiraldan** [comment]
> It looks like it's failing when calculating the experience for the first commit (`0/1966`), this made me think it could be caused by the loading of an earlier commit_experiences, but looking at your patch that doesn't seem to be possible...
> Could you check just to make sure that we hit
> 
> https://github.com/mozilla/bugbug/blob/cd6d8dc2c39355f46cb24b2f1e9386ce92ad47b0/bugbug/repository.py#L354
> and not
> 
> https://github.com/mozilla/bugbug/blob/cd6d8dc2c39355f46cb24b2f1e9386ce92ad47b0/bugbug/repository.py#L352
> ?

I will try to check

**Comment by Lothiraldan** [comment]
I'm not sure what changed since last time but I got a successful run before merging master:

```
+ bugbug-data-commits --limit 1000 cache
Prev version 3
Current version 3
Getting revs from -1000 to tip...
Mining 982 commits using 4 processes...
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5/5 [00:05<00:00,  1.19s/it]
Downloading file->component mapping...
ETAG MISMATCH None '"3c0faa7993f66ea9f1ffc18860a3b63f"'
Mining 982 commits using 4 processes...
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 982/982 [04:27<00:00,  3.67it/s]
Analyzing experiences from 982 commits...
EXPERIENCES FALLBACK!
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 982/982 [00:00<00:00, 487721.32it/s]
  0%|                                                                                                                                                                                        | 0/982 [00:00<?, ?it/s]Experience missing for file editor/libeditor/tests/browser_bug629172.js, type '', on commit 41c6beca5dc8c2d1ece11be076da2423a999f1c1
Experience missing for file editor/libeditor/tests/browser_bug629172.js, type 'backout', on commit 41c6beca5dc8c2d1ece11be076da2423a999f1c1
Experience missing for file browser/app/winlauncher/freestanding/DllBlocklist.cpp, type '', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file browser/app/winlauncher/freestanding/DllBlocklist.cpp, type 'backout', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file browser/app/winlauncher/freestanding/DllBlocklist.h, type '', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file browser/app/winlauncher/freestanding/DllBlocklist.h, type 'backout', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file mozglue/dllservices/Authenticode.cpp, type '', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file mozglue/dllservices/Authenticode.cpp, type 'backout', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file mozglue/dllservices/Authenticode.h, type '', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file mozglue/dllservices/Authenticode.h, type 'backout', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file mozglue/misc/WinUtils.h, type '', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file mozglue/misc/WinUtils.h, type 'backout', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file mozglue/dllservices/WindowsDllBlocklist.cpp, type '', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file mozglue/dllservices/WindowsDllBlocklist.cpp, type 'backout', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file mozglue/dllservices/WindowsDllBlocklist.h, type '', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file mozglue/dllservices/WindowsDllBlocklist.h, type 'backout', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file mozglue/dllservices/WindowsDllBlocklistCommon.h, type '', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file mozglue/dllservices/WindowsDllBlocklistCommon.h, type 'backout', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file mozglue/dllservices/WindowsDllBlocklistDefs.in, type '', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file mozglue/dllservices/WindowsDllBlocklistDefs.in, type 'backout', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file mozglue/dllservices/WindowsDllServices.h, type '', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file mozglue/dllservices/WindowsDllServices.h, type 'backout', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file mozglue/dllservices/gen_dll_blocklist_defs.py, type '', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file mozglue/dllservices/gen_dll_blocklist_defs.py, type 'backout', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file toolkit/components/telemetry/docs/data/third-party-modules-ping.rst, type '', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file toolkit/components/telemetry/docs/data/third-party-modules-ping.rst, type 'backout', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file toolkit/components/telemetry/tests/unit/test_ThirdPartyModulesPing.js, type '', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file toolkit/components/telemetry/tests/unit/test_ThirdPartyModulesPing.js, type 'backout', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file toolkit/xre/UntrustedModulesProcessor.cpp, type '', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file toolkit/xre/UntrustedModulesProcessor.cpp, type 'backout', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file toolkit/xre/UntrustedModulesProcessor.h, type '', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file toolkit/xre/UntrustedModulesProcessor.h, type 'backout', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file toolkit/xre/ModuleVersionInfo.cpp, type '', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file toolkit/xre/ModuleVersionInfo.cpp, type 'backout', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file toolkit/xre/ModuleVersionInfo.h, type '', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
Experience missing for file toolkit/xre/ModuleVersionInfo.h, type 'backout', on commit 2b745a11bb29bd2e4b56f13ab7e9ea211ffed3fb
  9%|████████████████                                                                                                                                                              | 91/982 [00:00<00:01, 822.19it/s]Experience missing for file tools/fuzzing/moz.build, type '', on commit b55c4b05eb3f49c66a63dddde3b0becea12d425d
Experience missing for file tools/fuzzing/moz.build, type 'backout', on commit b55c4b05eb3f49c66a63dddde3b0becea12d425d
Experience missing for file toolkit/components/search/tests/xpcshell/xpcshell.ini, type '', on commit 14789caee8bdde35e40a3bafa3ec7c786b8bc9f1
Experience missing for file toolkit/components/search/tests/xpcshell/xpcshell.ini, type 'backout', on commit 14789caee8bdde35e40a3bafa3ec7c786b8bc9f1
 29%|█████████████████████████████████████████████████▋                                                                                                                           | 282/982 [00:00<00:00, 886.16it/s]Experience missing for file browser/components/urlbar/tests/browser/browser_tip_keyboard_selection.js, type '', on commit e28a3e15c777095a9c55bb9e3bf42953c8e0e830
Experience missing for file browser/components/urlbar/tests/browser/browser_tip_keyboard_selection.js, type 'backout', on commit e28a3e15c777095a9c55bb9e3bf42953c8e0e830
 42%|████████████████████████████████████████████████████████████████████████▌                                                                                                    | 412/982 [00:00<00:00, 979.04it/s]Experience missing for file js/src/builtin/streams/ReadableStreamController.h, type '', on commit bd4746df4bfadf984f8aa6a509410f16fcc78303
Experience missing for file js/src/builtin/streams/ReadableStreamController.h, type 'backout', on commit bd4746df4bfadf984f8aa6a509410f16fcc78303
Experience missing for file third_party/rust/quote/tests/conditional/integer128.rs, type '', on commit 23803398111029a503a5ab228ad617bd0b9d728d
Experience missing for file third_party/rust/quote/tests/conditional/integer128.rs, type 'backout', on commit 23803398111029a503a5ab228ad617bd0b9d728d
Experience missing for file third_party/rust/unicode-xid/scripts/unicode.py, type '', on commit 23803398111029a503a5ab228ad617bd0b9d728d
Experience missing for file third_party/rust/unicode-xid/scripts/unicode.py, type 'backout', on commit 23803398111029a503a5ab228ad617bd0b9d728d
 60%|██████████████████████████████████████████████████████████████████████████████████████████████████████▊                                                                     | 587/982 [00:00<00:00, 1127.85it/s]Experience missing for file media/ffvpx/config_flac.h, type '', on commit 88679630171d6b735d02cdfbfb0a85ba182cee84
Experience missing for file media/ffvpx/config_flac.h, type 'backout', on commit 88679630171d6b735d02cdfbfb0a85ba182cee84
Experience missing for file build/build-clang/clang-8-android.json, type '', on commit de0fe40466cb3a451ba0680179cd6373d4d94baf
Experience missing for file build/build-clang/clang-8-android.json, type 'backout', on commit de0fe40466cb3a451ba0680179cd6373d4d94baf
Experience missing for file build/build-clang/clang-8-linux64-aarch64-cross.json, type '', on commit de0fe40466cb3a451ba0680179cd6373d4d94baf
Experience missing for file build/build-clang/clang-8-linux64-aarch64-cross.json, type 'backout', on commit de0fe40466cb3a451ba0680179cd6373d4d94baf
Experience missing for file build/build-clang/clang-8-linux64.json, type '', on commit de0fe40466cb3a451ba0680179cd6373d4d94baf
Experience missing for file build/build-clang/clang-8-linux64.json, type 'backout', on commit de0fe40466cb3a451ba0680179cd6373d4d94baf
Experience missing for file build/build-clang/clang-8-macosx64.json, type '', on commit de0fe40466cb3a451ba0680179cd6373d4d94baf
Experience missing for file build/build-clang/clang-8-macosx64.json, type 'backout', on commit de0fe40466cb3a451ba0680179cd6373d4d94baf
Experience missing for file taskcluster/scripts/misc/build-clang-8-linux-macosx-cross.sh, type '', on commit de0fe40466cb3a451ba0680179cd6373d4d94baf
Experience missing for file taskcluster/scripts/misc/build-clang-8-linux-macosx-cross.sh, type 'backout', on commit de0fe40466cb3a451ba0680179cd6373d4d94baf
Experience missing for file toolkit/components/extensions/ProxyScriptContext.jsm, type '', on commit d5ccb3bf4c1b2b6671d32e075a4a059a842cd9bd
Experience missing for file toolkit/components/extensions/ProxyScriptContext.jsm, type 'backout', on commit d5ccb3bf4c1b2b6671d32e075a4a059a842cd9bd
Experience missing for file toolkit/components/extensions/test/xpcshell/test_proxy_scripts_results.js, type '', on commit d5ccb3bf4c1b2b6671d32e075a4a059a842cd9bd
Experience missing for file toolkit/components/extensions/test/xpcshell/test_proxy_scripts_results.js, type 'backout', on commit d5ccb3bf4c1b2b6671d32e075a4a059a842cd9bd
Experience missing for file gfx/ots/src/cff_type2_charstring.cc, type '', on commit 81c77136eb578e19f946df1660e2ab2d3010df02
Experience missing for file gfx/ots/src/cff_type2_charstring.cc, type 'backout', on commit 81c77136eb578e19f946df1660e2ab2d3010df02
Experience missing for file gfx/ots/src/cff_type2_charstring.h, type '', on commit 81c77136eb578e19f946df1660e2ab2d3010df02
Experience missing for file gfx/ots/src/cff_type2_charstring.h, type 'backout', on commit 81c77136eb578e19f946df1660e2ab2d3010df02
Experience missing for file gfx/ots/tests/cff_type2_charstring_test.cc, type '', on commit 81c77136eb578e19f946df1660e2ab2d3010df02
Experience missing for file gfx/ots/tests/cff_type2_charstring_test.cc, type 'backout', on commit 81c77136eb578e19f946df1660e2ab2d3010df02
Experience missing for file toolkit/content/tests/browser/browser_block_plugIn.js, type '', on commit 263512ebd456125cdab2dab1a90bdaba65ee0137
Experience missing for file toolkit/content/tests/browser/browser_block_plugIn.js, type 'backout', on commit 263512ebd456125cdab2dab1a90bdaba65ee0137
Experience missing for file toolkit/content/tests/browser/browser_block_webAudio.js, type '', on commit 263512ebd456125cdab2dab1a90bdaba65ee0137
Experience missing for file toolkit/content/tests/browser/browser_block_webAudio.js, type 'backout', on commit 263512ebd456125cdab2dab1a90bdaba65ee0137
Experience missing for file third_party/rust/cranelift-codegen/src/ir/condcodes.rs, type '', on commit 7984123ec4f76cd753eaf6f3cff1bea3aa4878b9
Experience missing for file third_party/rust/cranelift-codegen/src/ir/condcodes.rs, type 'backout', on commit 7984123ec4f76cd753eaf6f3cff1bea3aa4878b9
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 982/982 [00:00<00:00, 1381.12it/s]
2019-10-24 10:14:41,417:INFO:scripts.commit_retriever:commit data extracted from repository
```

Relaunching after master and let's see if taskcluster is happy about it

**Comment by Lothiraldan** [comment]
> It looks like it's failing when calculating the experience for the first commit (`0/1966`), this made me think it could be caused by the loading of an earlier commit_experiences, but looking at your patch that doesn't seem to be possible...
> Could you check just to make sure that we hit
> 
> https://github.com/mozilla/bugbug/blob/cd6d8dc2c39355f46cb24b2f1e9386ce92ad47b0/bugbug/repository.py#L354
> and not
> 
> https://github.com/mozilla/bugbug/blob/cd6d8dc2c39355f46cb24b2f1e9386ce92ad47b0/bugbug/repository.py#L352
> ?

From my run before merging master, we were passing on line 354 as the file was not existing.

File: scripts/commit_retriever.py [hunk 1]
```diff
@@ -18,18 +18,26 @@ def __init__(self, cache_root):
...
+        if not db.is_old_version(repository.COMMITS_DB) and not limit:
             db.download(repository.COMMITS_DB, support_files_too=True)
 
             for commit in repository.get_commits():
                 pass
 
             rev_start = f"children({commit['node']})"
+
+            if limit:
+                logger.warning("Limit argument was ignored, the DB was used instead")
```
Comment (marco-c) [inline scripts/commit_retriever.py]: This should never happen because we have "and not limit"

File: .taskcluster.yml [hunk 2]
```diff
@@ -409,37 +409,37 @@ tasks:
...
-#                apt-get update &&
-#                apt-get install -y redis-server &&
-#                python -c 'import os; print(os.environ.keys())' &&
-#                bash ./scripts/integration_test.sh"
-#          metadata:
-#            name: bugbug integration test
-#            description: bugbug integration test
-#            owner: ${user}@users.noreply.github.com
-#            source: ${repository}/raw/${head_rev}/.taskcluster.yml
+      - $if: 'tasks_for == "github-push"'
```
Comment (marco-c) [inline .taskcluster.yml]: We should add a cache to this task like the commit retrieval task, so we don't clone the Mercurial repo from scratch all the time.

File: .taskcluster.yml [hunk 3]
```diff
@@ -409,37 +409,37 @@ tasks:
             owner: mcastelluccio@mozilla.com
             source: ${repository}/raw/${head_rev}/.taskcluster.yml
 
-#      - $if: 'tasks_for == "github-push" && head_branch[:10] == "refs/tags/"'
```
Comment (marco-c) [inline .taskcluster.yml]: Once it's ready, we need to put the second condition back.

File: scripts/commit_retriever.py (hunk 1 above)
Comment (Lothiraldan) [reply scripts/commit_retriever.py]: With the current code yes, do you want me to remove it or keep it so a future refactoring that might change when the limit is enforced or not would not have to add it back?

Comment (marco-c) [reply scripts/commit_retriever.py]: Let's remove it

File: infra/data-pipeline.yml [hunk 4]
```diff
@@ -718,44 +718,48 @@ tasks:
...
-#            apt-get update &&
-#            apt-get install -y redis-server &&
-#            python -c 'import os; print(os.environ.keys())' &&
-#            bash ./scripts/integration_test.sh"
-#      metadata:
-#        name: bugbug integration test
-#        description: bugbug integration test
-#        owner: release-mgmt-analysis@mozilla.com
-#        source: https://github.com/mozilla/bugbug/raw/master/data-pipeline.yml
+    - ID: integration-test
```
Comment (marco-c) [inline infra/data-pipeline.yml]: Add a comment saying this is the same as the one in .taskcluster.yml, so if we change one we don't forget to change the other

**Pull request merged**